chat/QuickJump.vue

```diff
@@ -0,0 +1,459 @@
+
+<script lang="ts">
+  import { Component, Watch } from '@f-list/vue-ts';
+  import Vue from 'vue';
+  import { getKey } from './common';
+  import core from './core';
+  import { Conversation } from './interfaces';
+  import { Keys } from '../keys';
+  import l from './localize';
+
+  interface SearchResult {
+    key: string;
+    name: string;
+    type: 'channel' | 'private' | 'console';
+    conversation?: Conversation;
+    description?: string;
+  }
+
+  @Component({})
+  export default class QuickJump extends Vue {
+    l = l;
+    visible = false;
+    searchQuery = '';
+    selectedIndex = 0;
+    allResults: SearchResult[] = [];
+
+    get filteredResults(): SearchResult[] {
+      if (this.searchQuery.length === 0) {
+        return this.allResults.slice(0, 10); // Show recent results when no query
+      }
+
+      const query = this.searchQuery.toLowerCase();
+      return this.allResults
+        .filter(
+          result =>
+            result.name.toLowerCase().includes(query) ||
+            (result.description &&
+              result.description.toLowerCase().includes(query))
+        )
+        .slice(0, 10); // Limit to 10 results
+    }
+
+    show(): void {
+      this.visible = true;
+      this.updateResults();
+      this.searchQuery = '';
+      this.selectedIndex = 0;
+      this.$nextTick(() => {
+        const input = this.$refs.searchInput as HTMLInputElement;
+        if (input) {
+          input.focus();
+        }
+      });
+    }
+
+    hide(): void {
+      this.visible = false;
+      this.searchQuery = '';
+      this.selectedIndex = 0;
+    }
+
+    updateResults(): void {
+      const results: SearchResult[] = [];
+
+      // Add console tab
+      results.push({
+        key: 'console',
+        name: core.conversations.consoleTab.name,
+        type: 'console',
+        conversation: core.conversations.consoleTab,
+        description: l('quickJump.consoleDescription')
+      });
+
+      // Add private conversations
+      for (const conversation of core.conversations.privateConversations) {
+        results.push({
+          key: conversation.key,
+          name: conversation.name,
+          type: 'private',
+          conversation,
+          description: l('quickJump.privateConversation')
+        });
+      }
+
+      // Add channel conversations
+      for (const conversation of core.conversations.channelConversations) {
+        results.push({
+          key: conversation.key,
+          name: conversation.name,
+          type: 'channel',
+          conversation,
+          description: `${conversation.channel.sortedMembers.length} ${l('quickJump.members')}`
+        });
+      }
+
+      // Sort by recent activity (unread first, then by last message time)
+      results.sort((a, b) => {
+        if (!a.conversation || !b.conversation) return 0;
+
+        // Prioritize conversations with unread messages
+        if (
+          a.conversation.unread !== Conversation.UnreadState.None &&
```

Comment:
This makes a lot of sense to do. Maybe it'd be even better if the search result also showed why certain conversations are being prioritised. Could be something as simple as a badge (which has a bootstrap class already 😉).